x-pack/plugins/observability_solution/synthetics/server/synthetics_service/authentication/check_has_privilege.ts

```diff
@@ -9,24 +9,27 @@ import { SecurityIndexPrivilege } from '@elastic/elasticsearch/lib/api/types';
 import { UptimeEsClient } from '../../lib';
 import { SyntheticsServerSetup } from '../../types';
 import { getFakeKibanaRequest } from '../utils/fake_kibana_request';
-import { serviceApiKeyPrivileges, syntheticsIndex } from '../get_api_key';
+import { getServiceApiKeyPrivileges, syntheticsIndex } from '../get_api_key';
 
-export const checkHasPrivileges = async (
+export const checkHasPrivileges = (
   server: SyntheticsServerSetup,
   apiKey: { id: string; apiKey: string }
 ) => {
-  return await server.coreStart.elasticsearch.client
+  const { indices: index, cluster } = getServiceApiKeyPrivileges(
+    server.coreStart.elasticsearch.getCapabilities().serverless
```

Comment:
@justinkambic could we keep exposing the `isServerless` in the `SyntheticsServerSetup` definition? this is a long chain of functions to remember to check if it serverless or not.

Reply:
@cauemarcondes the Kibana team stressed that Kibana's `buildFlavor` and `es capabilities` (which is basically Elasticsearch's build flavor) should be treated as two separate things. It's a bit pedantic but the two aren't synonymous and there's no way to know if there will be important differences in the future.

I agree that this implementation isn't great because of the lengthy call chain, and it's not DRY, so we should improve it as you suggest. I can add an `isEsServerless` flag to the plugin class and reference it in the server object that gets injected in place of this. 👍

Reply:
https://github.com/elastic/kibana/pull/178897/commits/30ddac0e9a19a9b0abaa3b2dcc444450ddae03e6